### ISSUE TITLE
ApiCompatibility Symbol filtering story is incomplete

### ISSUE THREAD

**Issue by safern** [dotnet/sdk#16920]
We are missing Attribute filtering and some scenarios in the accessibility filter, see: https://github.com/dotnet/sdk/pull/16817#discussion_r613366572

For Attribute filtering I think it makes sense to have a filter dedicated to it that ignores compiler generated attributes and a default list of attributes that shouldn't contribute to API Compatibility like `EditorBrowsableAttribute`. Also it should be able to take a list of attributes to ignore based on user settings.

cc: @ericstj @Anipik

**Comment by ViktorHofer** [comment]
@ericstj regarding [your comment](https://github.com/dotnet/sdk/pull/16817#discussion_r613366572):

> Take note of this scenario: https://github.com/dotnet/arcade/blob/3dd26504b456b68c2e3edd18ebf159d0b5486135/src/Microsoft.Cci.Extensions/Filters/PublicOnlyCciFilter.cs#L55-L66
Maybe those are other filters but they did matter for our existing API compat and GenAPI analysis.

I did take a look to check if we are unintentionally excluding API that should be checked. In the example above, the filter checks if a member is not public but abstract and is part of a public abstract type. At least for C# the compiler emits an error as "virtual or abstract members cannot be private":

<img width="492" alt="image" src="https://user-images.githubusercontent.com/7412651/185784872-102fec7c-7a3c-4587-87a7-aea2a27e9f51.png">

I would also need to check if the same is true for VB and F# but at least just for C# this logic doesn't seem to be necessary.

**Comment by ericstj** [comment]
Try that with `internal` instead of `private`.   That case was there specifically because we had types which were defining `internal abstract` members.

**Comment by ViktorHofer** [comment]
Got it. Found the related issues: https://github.com/dotnet/runtime/issues/28919 and https://github.com/dotnet/arcade/pull/2650 which provide more information. While these issues are about emitting internal abstract members in reference source files via GenAPI, do we also care about validating them via APICompat? I would probably say yes but maybe only when [`IncludeInternalSymbols` is set to true](https://github.com/dotnet/sdk/blob/5f19e30aec7c40978ca677d99b3ed5f84b097c34/src/ApiCompat/Microsoft.DotNet.ApiCompatibility/ApiComparerSettings.cs#L26)?
